Source/API/genetic_operator.h

```diff
@@ -3,7 +3,7 @@
 
 class genetic_operator {
 public:
-    virtual double get_weight() = 0;
+    virtual double get_weight() const = 0;
```

Comment:
Shold this be `default` instead of `= 0`?

Reply:
Your right default makes more sense for readability. Do you think this can wait till the pointer audit branch?

Reply:
looks like we can only make constructor/destructors `default`